pg_auth_mon.c

```diff
@@ -141,17 +151,22 @@ _PG_init(void)
 						NULL);
 }
 
+#if PG_VERSION_NUM >= 150000
 /*
- * Module unload callback
+ * shmem_request hook: request additional shared resources.  We'll allocate or
+ * attach to the shared resources in fai_shmem_startup().
  */
-void
-_PG_fini(void)
+static void
+fai_shmem_request(void)
 {
-	/* Uninstall hooks. */
-	shmem_startup_hook = prev_shmem_startup_hook;
-	ClientAuthentication_hook = original_client_auth_hook;
-
+	RequestAddinShmemSpace(fai_memsize());
+#if PG_VERSION_NUM < 90600
```

Comment:
this condition is mututally exclusive with the `if PG_VERSION_NUM >= 150000` - [line 154 ](https://github.com/sdudoladov/pg_auth_mon/pull/2/files#diff-b6d96da0f687f3f578be83b08ce801356890648257ca815daf165d0002d3d70eR154)

Reply:
Oh, I don't even know how this code ended up here. Thank you very much.